### DISCUSSION TITLE
Performance Benchmarking of Polarfire SOC to other ARM processors

### DISCUSSION THREAD

**Discussion by josh25219** [polarfire-soc/polarfire-soc-documentation#325] [General] [answered]
Hi,

For someone considering using a RISCV quad core processor such as Polarfire SOC, what is the best way to compare the performance of Polarfire SOC to other ARM solutions for specific use case scenarios?   Since performance can be tied to the configuration of the memory subsystem,  Is Renode a viable option for doing Coremark type measurements across various processors?

Thanks,

**Comment by hughbreslin** [comment, marked as answer]
Hi @josh25219 

I wouldn't say renode is an accurate benchmark as it is just emulating the CPU, it isn't cycle accurate so it wouldn't give you a clear result. Your best bet is to benchmark on one of our development kits for the configuration you would like to use.

We have a benchmarking whitepaper which may be useful for you - you can find a copy [here](https://ww1.microchip.com/downloads/aemDocuments/documents/FPGA/ProductDocuments/SupportingCollateral/whitepapers/Microchip_PolarFire_SoC_CPU_Performance_Benchmarking_White_Paper+%282%29.pdf) - this gives you results in bare metal for Dhrystone and Coremark running from multiple memories and also Coremark Pro on Linux.

The easiest way to think about it is that 2 of our cores (we have 4 application cores) are equivalent to an ARM A9 - so our quad application core system is equivalent to 2x A9s

**Reply by hughbreslin** [reply]
And one thing to add - we do provide a sample application for testing coremark in different configurations across the cores, you can find it [here](https://github.com/polarfire-soc/polarfire-soc-bare-metal-examples/tree/main/driver-examples/mss/mpfs-hal/mpfs-hal-coremark)
